code/__HELPERS/test_reach.dm

```diff
@@ -0,0 +1,128 @@
+//test_reach(), by Deity Link
+//Basically sends a cheap projectile that moves from a turf A to a turf B in a "straight" line. Returns true if it reaches turf B without hitting anything on the way there (other than turf B itself)
+//Useful for instance to check if an item thrown/projectile fired from a A to B would reach its target
+
+/proc/test_reach(var/turf/origin,var/turf/destination,var/_pass_flags=0)
+	if (!origin || !destination)
+		return FALSE
+	if (origin.z != destination.z)
+		return FALSE
+	if (origin == destination)
+		return TRUE
+	var/obj/test_reach/test = new(origin,destination,_pass_flags)
+	var/result = test.main()
+	qdel(test)
+	return result
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+/obj/test_reach
```

Comment:
I've actually had a similar problem to deal with in my euro lights PR. Vector objects/datums solve this in a better way, I think; they don't come with the overhead of creating an atom.

The point I'm trying to make is please please use vectors/raycasts. Thx.
vector_procs.dm already has a bunch of helpers.

Reply:
Trust me: my brain works at 5% speed when trying to use vectors in Byond, and I don't think it makes the code more maintainable. Sorry, I won't do it. You're free to try and convert this to use vectors in the future if you manage to reach the exact same behaviour. In the case of splashing paint the test would be something like this: 
![image](https://github.com/vgstation-coders/vgstation13/assets/7573912/f75fc546-9ec8-41e3-be77-cc793a573ae7)

Reply:
That took me hours until I had the idea to make the first step manually avoid walls if possible.

Reply:
I mean I understand but this I can guarantee there's a much better way of handling this. Probably faster as well. If/when I got the time I'll get around to it.